pkg/isoeditor/ignition_image.go

```diff
@@ -0,0 +1,68 @@
+package isoeditor
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+)
+
+type FileData struct {
+	Filename string
+	Data     io.ReadCloser
+}
+
+type ignitionImageReader struct {
+	io.Reader
+	io.Closer
+}
+
+// NewIgnitionImageReader returns the filename of the ignition image in the ISO,
+// along with a stream of the ignition image with ignition content embedded.
+// This can be used to overwrite the ignition image file of an ISO previously
+// unpacked by Extract() in order to embed ignition data.
+func NewIgnitionImageReader(isoPath string, ignitionContent *IgnitionContent) ([]FileData, error) {
```

Comment:
Looks good, but we should probably add a test of some sort for this.

Reply:
Done